pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -41,45 +41,33 @@ const (
 	ManagementClusterHostname = "mgmtCluster"
 )
 
-// TODO: Need to do inheritance of DestRules based on domain suffix match
-
 // BuildClusters returns the list of clusters for the given proxy. This is the CDS output
 // For outbound: Cluster for each service/subset hostname or cidr with SNI set to service hostname
 // Cluster type based on resolution
 // For inbound (sidecar only): Cluster for each inbound endpoint port and for each service port
 func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, proxy *model.Proxy, push *model.PushContext) ([]*v2.Cluster, error) {
 	clusters := make([]*v2.Cluster, 0)
 
-	recomputeOutboundClusters := true
```

Comment:
Can we keep this code - most users will not use Sidecar - and I want to avoid destabilizing their code or adding regressions. It is too risky to use the new code.

Reply:
You said the CDS cache is not that effective. If thats the case, then this code is simply removing the cache. There is actually nothing here that relates to using Sidecar objects

Reply:
CDS cache is super effective - I said CDS optimizations (subsetting) are low priority compared with LDS/RDS exactly because the cache is so effective.

Reply:
And even if cache would not be effective - no good reason to change how it works for non-Sidecar users. This PR is supposed to support Sidecar, not change non-sidecar behavior